File: ios/BlakJaks/Networking/APIEndpoints.swift
import Foundation

// MARK: - APIEndpoints
// Centralized endpoint definitions. All paths are relative to Config.apiBaseURL.

enum APIEndpoints {

    // MARK: - Auth
    static let login             = "/auth/login"
    static let signup            = "/auth/register"
    static let logout            = "/auth/logout"
    static let refreshToken      = "/auth/refresh"
    static let intercomToken     = "/auth/intercom-token"

    // MARK: - User
    static let me                = "/users/me"
    static let memberCard        = "/users/me/member-card"
    static let avatarUpload      = "/users/me/avatar"

    // MARK: - Insights
    static let insightsOverview  = "/insights/overview"
    static let insightsTreasury  = "/insights/treasury"
    static let insightsSystems   = "/insights/systems"
    static let insightsComps     = "/insights/comps"
    static let insightsPartners  = "/insights/partners"
    static let insightsFeed      = "/insights/feed"

    // MARK: - Scan & Wallet
    static let scan              = "/scans"
    static let scanHistory       = "/scans/history"
    static let wallet            = "/wallet"
    static let transactions      = "/wallet/transactions"
    static let compVault         = "/wallet/comp-vault"
    static let cryptoWithdraw    = "/wallet/withdraw/crypto"
    static let bankWithdraw      = "/wallet/withdraw/bank"

    // MARK: - Dwolla ACH
    static let dwollaFundingSources = "/dwolla/funding-sources"
    static let dwollaPlaidToken     = "/dwolla/plaid/link-token"
    static let dwollaExchangeSession = "/dwolla/exchange-sessions"
    static func dwollaFundingSource(_ id: String) -> String { "/dwolla/funding-sources/\(id)" }
    static let dwollaTransfers      = "/dwolla/transfers"
    static func dwollaTransfer(_ id: String) -> String { "/dwolla/transfers/\(id)" }

    // MARK: - Shop
    static let products          = "/shop/products"
    static func product(_ id: Int) -> String { "/shop/products/\(id)" }
    static let cart              = "/shop/cart"
    static let cartItems         = "/shop/cart/items"
    static func cartItem(_ id: Int) -> String { "/shop/cart/items/\(id)" }
    static let taxEstimate       = "/shop/tax-estimate"
    static let orders            = "/shop/orders"
    static func order(_ id: Int) -> String { "/shop/orders/\(id)" }
    static let checkout          = "/shop/checkout"

    // MARK: - Notifications
    static let notifications         = "/notifications"
    static let notificationsMarkRead = "/notifications/read-all"
    static let pushToken             = "/notifications/push-token"

    // MARK: - Social Hub
    static let channels          = "/social/channels"
    static func channelMessages(_ id: String) -> String { "/social/channels/\(id)/messages" }
    static let liveStreams        = "/social/streams"

    // MARK: - Governance
    static let proposals         = "/governance/proposals"
    static func vote(_ id: Int) -> String { "/governance/proposals/\(id)/vote" }

    // MARK: - Wholesale / Affiliate
    static let wholesaleDashboard  = "/wholesale/dashboard"
    static let affiliateDashboard  = "/affiliate/dashboard"
    static let affiliateReferrals  = "/affiliate/referrals"
    static let affiliatePayouts    = "/affiliate/payouts"
}

File: android/app/src/main/java/com/blakjaks/app/core/network/ApiEndpoints.kt
package com.blakjaks.app.core.network

object ApiEndpoints {
    // Auth
    const val LOGIN = "/auth/login"
    const val SIGNUP = "/auth/signup"
    const val LOGOUT = "/auth/logout"
    const val REFRESH_TOKEN = "/auth/refresh"
    const val INTERCOM_TOKEN = "/intercom/token"

    // User
    const val ME = "/users/me"
    const val UPDATE_PROFILE = "/users/me"
    const val UPLOAD_AVATAR = "/users/me/avatar"
    const val MEMBER_CARD = "/users/me/member-card"
    const val WALLET = "/users/me/wallet"
    const val PUSH_TOKEN = "/users/me/push-token"

    // Insights
    const val INSIGHTS_OVERVIEW = "/insights/overview"
    const val INSIGHTS_TREASURY = "/insights/treasury"
    const val INSIGHTS_SYSTEMS = "/insights/systems"
    const val INSIGHTS_COMPS = "/insights/comps"
    const val INSIGHTS_PARTNERS = "/insights/partners"
    const val INSIGHTS_FEED = "/insights/feed"

    // Scan & Wallet
    const val SUBMIT_SCAN = "/scans/submit"
    const val SCAN_HISTORY = "/scans/history"
    const val WALLET_TRANSACTIONS = "/wallet/transactions"
    const val COMP_VAULT = "/users/me/comp-vault"
    const val WITHDRAW = "/wallet/withdraw"
    const val COMP_PAYOUT_CHOICE = "/wallet/comp-payout-choice"

    // Dwolla
    const val DWOLLA_FUNDING_SOURCES = "/users/me/dwolla/funding-sources"
    const val DWOLLA_CREATE_CUSTOMER = "/users/me/dwolla/customer"
    const val PLAID_LINK_TOKEN = "/users/me/dwolla/plaid-link-token"
    const val LINK_BANK = "/users/me/dwolla/link-bank"
    const val WITHDRAW_TO_BANK = "/users/me/dwolla/withdraw"

    // Shop
    const val PRODUCTS = "/shop/products"
    const val CART = "/cart"
    const val ADD_TO_CART = "/cart/add"
    const val UPDATE_CART_ITEM = "/cart/update"
    const val ESTIMATE_TAX = "/tax/estimate"
    const val CREATE_ORDER = "/orders/create"

    // Notifications
    const val NOTIFICATIONS = "/notifications"
    const val MARK_ALL_READ = "/notifications/read-all"
    const val UNREAD_COUNT = "/notifications/unread-count"

    // Social
    const val CHANNELS = "/social/channels"
    const val SEND_MESSAGE = "/social/channels/{channelId}/messages"
    const val TRANSLATE_MESSAGE = "/social/messages/{id}/translate"
    const val ADD_REACTION = "/social/messages/{id}/reactions"

    // Affiliate
    const val AFFILIATE_DASHBOARD = "/affiliate/dashboard"
    const val AFFILIATE_PAYOUTS = "/affiliate/payouts"
    const val AFFILIATE_REFERRAL_CODE = "/affiliate/referral-code"
}
